hedera-node/src/main/java/com/hedera/services/txns/crypto/validators/AdjustAllowanceChecks.java

```diff
@@ -109,30 +128,37 @@ public ResponseCodeEnum validateCryptoAllowances(final List<CryptoAllowance> cry
 
 	@Override
 	public ResponseCodeEnum validateFungibleTokenAllowances(final List<TokenAllowance> tokenAllowancesList,
-			final Account ownerAccount) {
+			final Account payerAccount) {
 		if (tokenAllowancesList.isEmpty()) {
 			return OK;
 		}
 		final var tokenKeys = tokenAllowancesList
 				.stream()
-				.map(a -> FcTokenAllowanceId.from(EntityNum.fromTokenId(a.getTokenId()),
-						EntityNum.fromAccountId(a.getSpender())))
+				.map(a -> buildTokenAllowanceKey(a.getOwner(), a.getTokenId(), a.getSpender()))
 				.toList();
 		if (hasRepeatedId(tokenKeys)) {
 			return SPENDER_ACCOUNT_REPEATED_IN_ALLOWANCES;
 		}
 
-		final var existingAllowances = ownerAccount.getFungibleTokenAllowances();
-
 		for (final var allowance : tokenAllowancesList) {
-			final var spenderAccountId = allowance.getSpender();
 			final var owner = Id.fromGrpcAccount(allowance.getOwner());
+			final var spender = Id.fromGrpcAccount(allowance.getSpender());
 			final var amount = allowance.getAmount();
 			final var tokenId = allowance.getTokenId();
 			final var token = tokenStore.loadPossiblyPausedToken(Id.fromGrpcToken(tokenId));
-			final var spenderId = Id.fromGrpcAccount(spenderAccountId);
+			final Account ownerAccount;
+			if (owner.equals(Id.MISSING_ID) || owner.equals(payerAccount.getId())) {
```

Comment:
The L150-158 block is duplicated and should be fairly easy to extract and re-use I think.